### PR TITLE
Retrive Videos

### PR DESCRIPTION
**Pull request by TheRealPikachu**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by RigidStudios** [comment]
> **Please describe the changes this PR makes and why it should be merged:**

_\*cricket noises\*_

File: actions/retrieve_videos_MOD.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  action: async function (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const id = this.evalMessage(data.id, cache)
+    const Mods = this.getMods()
+    const ytch = Mods.require('yt-channel-info')
+    const channelId = id
+    const sortBy = 'newest'
```
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:114]: You could actually include a field for the sorting to make the mod more versatile. The three sort modes are `newest`, `oldest`, `popular`.

File: actions/retrieve_videos_MOD.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  // ---------------------------------------------------------------------
+
+  action: async function (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const id = this.evalMessage(data.id, cache)
+    const Mods = this.getMods()
+    const ytch = Mods.require('yt-channel-info')
+    const channelId = id
```
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:113]:
```suggestion
    const channelId = id
```

File: actions/retrieve_videos_MOD.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  // This is the function for the action within the Bot's Action class.
+  // Keep in mind event calls won't have access to the "msg" parameter,
+  // so be sure to provide checks for variable existance.
+  // ---------------------------------------------------------------------
+
+  action: async function (cache) {
+    const data = cache.actions[cache.index]
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const id = this.evalMessage(data.id, cache)
```
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:110]:
```suggestion
    const channelId = this.evalMessage(data.id, cache)
```

File: actions/retrieve_videos_MOD.js (hunk 2 above)
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:113]:
```suggestion

```

File: actions/retrieve_videos_MOD.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    const sortBy = 'newest'
+
+    const videos = await ytch.getChannelVideos(channelId, sortBy)
+    const items = videos.items
+    const finalizedresults = JSON.stringify(items)
+
+    this.storeValue(finalizedresults, storage, varName, cache)
+    this.callNextAction(cache)
+  },
+  // ---------------finalresults--------------------------------------------------
```
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:123]:
```suggestion
  // -----------------------------------------------------------------
```

File: actions/retrieve_videos_MOD.js (hunk 3 above)
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:110]:
```suggestion
    const channelId = this.evalMessage(data.id, cache)
```

File: actions/retrieve_videos_MOD.js (hunk 2 above)
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:113]:
```suggestion

```

File: actions/retrieve_videos_MOD.js (hunk 4 above)
Comment (RigidStudios) [inline actions/retrieve_videos_MOD.js:123]:
```suggestion
  // -----------------------------------------------------------------
```

**Comment by RigidStudios** [comment]
Also, before Alm tells you to, add Documentation for this mod to this branch.

**Comment by almeidx** [comment]
This could just be added to https://github.com/dbm-network/mods/pull/493

**Comment by noahalma** [comment]
lol

**Pull request closed without merging**